pkg/broker/handler/pool/retry/pool.go

```diff
@@ -50,6 +50,25 @@ type SyncPool struct {
 	deliverClient ceclient.Client
 }
 
+type handlerCache struct {
```

Comment:
I guess there is already a plan to consolidate fanout pool and retry pool since they share a lot in common?

Reply:
We have extracted some common code to pkg/broker/handler/pool/pool.go. It's not very obvious to me how to share the remaining code. Feel free to share your thoughts.

For `handlerCache` it may seem that you can just store topic/sub (vs. broker and target). But that may not be the only thing we check to modify handlers. E.g. we may need to check state to stop some handlers.

Reply:
Just at a glance there are some similarities but haven't read carefully. I will give it a try and see if it makes sense to consolidate more :)

Let's keep this PR as is.